applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c

```diff
@@ -4,3 +4,293 @@
+
+	/* We only support one QoS at the moment, reject changes */
+	return -ENOEXEC;
+}
+
+static int lc3_cap_qos_cb(struct bt_audio_stream *stream, struct bt_codec_qos *qos)
+{
+	LOG_INF("QoS: stream %p qos %p", (void *)stream, (void *)qos);
+
+	return 0;
+}
+
+static int lc3_cap_enable_cb(struct bt_audio_stream *stream, struct bt_codec_data *meta,
+			     size_t meta_count)
+{
+	int ret;
+	struct event_t event;
+
+	LOG_INF("Enable: stream %p meta_count %u", (void *)stream, meta_count);
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+	return 0;
+}
+
+static int lc3_cap_start_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Start: stream %p", (void *)stream);
+	return 0;
+}
+
+static int lc3_cap_metadata_cb(struct bt_audio_stream *stream, struct bt_codec_data *meta,
+			       size_t meta_count)
+{
+	LOG_INF("Metadata: stream %p meta_count %u", (void *)stream, meta_count);
+	return 0;
+}
+
+static int lc3_cap_disable_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Disable: stream %p", (void *)stream);
+	return 0;
+}
+
+static int lc3_cap_stop_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	LOG_INF("Stop: stream %p", (void *)stream);
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_NOT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+
+	return 0;
+}
+
+static int lc3_cap_release_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Release: stream %p", (void *)stream);
+	return 0;
+}
+static struct bt_audio_capability_ops lc3_cap_codec_ops = {
+	.config = lc3_cap_config_cb,
+	.reconfig = lc3_cap_reconfig_cb,
+	.qos = lc3_cap_qos_cb,
+	.enable = lc3_cap_enable_cb,
+	.start = lc3_cap_start_cb,
+	.metadata = lc3_cap_metadata_cb,
+	.disable = lc3_cap_disable_cb,
+	.stop = lc3_cap_stop_cb,
+	.release = lc3_cap_release_cb,
+};
+
+static void stream_recv_cb(struct bt_audio_stream *stream, const struct bt_iso_recv_info *info,
+			   struct net_buf *buf)
+{
+	static uint32_t recv_cnt;
+	bool bad_frame = false;
+
+	if (receive_cb == NULL) {
+		LOG_ERR("The RX callback has not been set");
+		return;
+	}
+
+	if (!(info->flags & BT_ISO_FLAGS_VALID)) {
+		bad_frame = true;
+	}
+
+	receive_cb(buf->data, buf->len, bad_frame, info->ts);
+
+	recv_cnt++;
+	if ((recv_cnt % 1000U) == 0U) {
+		LOG_INF("Received %u total ISO packets", recv_cnt);
+	}
+}
+
+static void stream_stop_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	LOG_INF("Stop: stream %p", (void *)stream);
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_NOT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+}
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
```

Comment:
This should still be moved below 
```C
if (err) {
    default_conn = NULL;
    return;
}
```
No need to call this function if there is an err